src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -351,10 +351,7 @@ export function getDiscoverStateContainer({
     const unsubscribeData = dataStateContainer.subscribe();
 
     // updates saved search when query or filters change, triggers data fetching
-    const filterUnsubscribe = merge(
-      services.data.query.queryString.getUpdates$(),
```

Comment:
If the query changed, this is also propagated to AppState, which now also handles triggering the refetch in this case. there were too many ways to trigger fetching for query changes. less is more in this case, it's much easier to debug

Reply:
query change logic is now handled here: https://github.com/elastic/kibana/pull/159903/files#r1236803970